servers/src/main/java/tachyon/worker/netty/DataServerHandler.java

```diff
@@ -59,14 +89,15 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) throw
 
     BlockHandler handler = null;
     try {
-      validateInput(req);
+      req.validate();
       handler = storageDir.getBlockHandler(blockId);
 
       final long fileLength = handler.getLength();
       validateBounds(req, fileLength);
       final long readLength = returnLength(offset, len, fileLength);
       ChannelFuture future =
-          ctx.writeAndFlush(new BlockResponse(blockId, offset, readLength, handler));
+          ctx.writeAndFlush(new RPCBlockResponse(blockId, offset, readLength, getDataBuffer(req,
+              handler, (int) readLength)));
```

Comment:
Is length bounded by integer max? Previously the netty file region object still used a long?